lib/api/strapi.ts

```diff
@@ -0,0 +1,133 @@
+import qs from 'qs'
+
+export const API_URL = `${process.env.NEXT_PUBLIC_BASE_URL}/api`
+
+export type StrapiFetchOptions = {
+  query?: string
+  url?: string
+}
+
+export type SingleResponse<T> = {
+  id: number
+  attributes: Omit<T, 'id'>
+}
+
+export type CollectionResponse<T> = SingleResponse<T>[]
+
+type StrapiError = {
+  status: number
+  name: string
+  message: string
+  details: unknown
+}
+
+export type PagePagination = {
+  page?: number
+  pageSize?: number
+  withCount?: boolean
+}
+
+export type OffsetPagination = {
+  start?: number
+  limit?: number
+  withCount?: boolean
+}
+
+export type Pagination = PagePagination | OffsetPagination
+
+export type Meta<T extends Pagination> = T extends PagePagination
+  ? {
+      pagination: {
+        page: number
+        pageSize: number
+        pageCount: number
+        total?: number
+      }
+    }
+  : {
+      pagination: {
+        start: number
+        limit: number
+        total?: number
+      }
+    }
+
+type StrapiResponse<
+  T,
+  S extends SingleResponse<T> | CollectionResponse<T> = CollectionResponse<T>,
+  P extends Pagination = PagePagination
+> = {
+  data: S | null
+  error?: StrapiError
+  meta: Meta<P>
+}
+
+export async function fetchSingle<T>(
+  path: string,
+  options?: Omit<StrapiFetchOptions, 'slug'>
+): Promise<StrapiResponse<T, SingleResponse<T>> | null> {
+  const url = options?.url ?? API_URL
+  const query = options?.query ?? ''
+
+  return fetchFromStrapi<T, SingleResponse<T>>(`${url}${path}?${query}`)
+}
+
+export async function fetchCollectionSingle<T>(
+  path: string,
+  slug: string,
+  options?: StrapiFetchOptions
+): Promise<StrapiResponse<T, SingleResponse<T>> | null> {
+  const url = options?.url ?? API_URL
+  const query = options?.query ?? ''
+  const slugQuery = qs.stringify({
+    filters: {
+      slug: {
+        $eq: slug,
+      },
+    },
+  })
+
+  const res = await fetchFromStrapi<T, CollectionResponse<T>>(
+    `${url}${path}?${query}&${slugQuery}`
+  )
+
+  if (!res?.data?.[0]) {
+    return null
+  }
+
+  return {
+    ...res,
+    data: res?.data?.[0] ?? null,
```

Comment:
```suggestion
    data: res.data[0]
```
res.data[0] is already checked for null value above